Add "robot" container

amarburg opened the pull request:
## Changes Made

The existing Dockerfile defines a "robot" stage which contains the Blue source and some deps but no other source files (from `blue.repos`).   This means "robot" is not runnable "out of the box," requires multiple manual stages to make it runnable.

This PR:

* Renames the "robot" stage as "blue_src_and_deps"
* Updates "robot" to include all dependencies of "blue" and "colcon build"

## Testing

Ongoing.

amarburg commented:
CI is broken on this until #220 (Rolling is now based on Noble) is resolved

evan-palmer commented:
I like the idea of adding making `robot` usable out-of-the-box. We probably don't need to split the two into separate stages, though. That leaves the `blue_src_and_deps` as unusable. Regarding splitting the `.repos` files up, that was actually what I used to do, before merging them; [Angler](https://github.com/Robotic-Decision-Making-Lab/angler) still does that. I named them `blue.repos` and `sim.repos`. I merged them to help reduce the number of files to maintain, but I'm open to splitting them up for branches that don't have supported binaries. I do also like the idea of a `robot` devcontainer. Maybe we could add the remote development extension by default to that version?

amarburg commented:
- I preserved `blue_src_and_deps` as I didn't know if it was used.   Sounds like no ... I've removed it
- Renamed to `.repos` --> `blue.repos` and `sim.repos`.   Sim.repos may be changed or disappear depending on how this Gazebo realignment goes.

```
Maybe we could add the remote development extension by default to that version?
```

- Don't know enough about VSCode development containers to know what this means.

Getting clean builds for all of the different branches requires #220 and #223

evan-palmer commented on .docker/Dockerfile:
```diff
@@ -36,6 +37,14 @@ RUN apt-get -q update \
     && apt-get clean -y \
     && rm -rf /var/lib/apt/lists/*
 
+# This stage includes
```
Thanks for adding these docs. This is helpful.

evan-palmer reviewed (approved):
Looks good! Thanks for adding this

evan-palmer commented:
@mergify backport humble iron
